hybris/common/hooks.c

```diff
@@ -2935,21 +3092,26 @@ static void* __hybris_get_hooked_symbol(const char *sym, const char *requester)
 
     if (!sorted)
     {
+        qsort(hooks_properties, HOOKS_SIZE(hooks_properties), sizeof(hooks_properties[0]), hook_cmp);
         qsort(hooks_common, HOOKS_SIZE(hooks_common), sizeof(hooks_common[0]), hook_cmp);
         qsort(hooks_mm, HOOKS_SIZE(hooks_mm), sizeof(hooks_mm[0]), hook_cmp);
         sorted = 1;
     }
 
     /* Allow newer hooks to override those which are available for all versions */
     key.name = sym;
-#if defined(WANT_LINKER_N)
-    if (get_android_sdk_version() > 21)
-        found = bsearch(&key, hooks_mm, HOOKS_SIZE(hooks_mm), sizeof(hooks_mm[0]), hook_cmp);
-#endif
-#if defined(WANT_LINKER_MM)
-    if (get_android_sdk_version() > 21)
+    sdk_version = get_android_sdk_version();
+
+#if defined(WANT_LINKER_MM) || defined(WANT_LINKER_N) || defined(WANT_LINKER_O)
+    if (sdk_version > 21)
         found = bsearch(&key, hooks_mm, HOOKS_SIZE(hooks_mm), sizeof(hooks_mm[0]), hook_cmp);
 #endif
+    // make sure to skip the property hooks only when o.so is actually loaded
+    // since for testing and we sometimes set things like 99 as sdk version.
+    // The o linker is loaded when sdk_version >= 27 and exists.
+    if (!found && sdk_version < 24)
```

Comment:
Shouldn't it be < 27, here ? Otherwise it won't work with N linker.

Reply:
this is related to https://github.com/libhybris/libhybris/pull/407/commits/bae4f72d935cdda72479b8d02285a8fdb6b8b98e

i need to wrap my head around this but basically it should work because of this: 
https://github.com/libhybris/libhybris/pull/407/commits/bae4f72d935cdda72479b8d02285a8fdb6b8b98e#diff-cf1210eb0ad9879f7d812c63baf11fb4R3386

Reply:
@tofe can you check the output of "/usr/bin/getprop ro.build.version.sdk"

Reply:
It returns nothing in with "< 24", and it returns "25" with my little patching

Reply:
i have reverted the patch since it breaks compatibility with halium

it's currently necessary on android 8 adaptations so i kept it for those, but we can look at that issue later